code/modules/mob/observer/freelook/marker/signal_abilities/psychic_tracer.dm

```diff
@@ -133,11 +135,11 @@
 	remove_extension(holder, base_type)
 
 /datum/extension/psychic_tracer/Destroy()
-	QDEL_NULL(object)
```

Comment:
QDEL_NULL isnt a hard delete, it just calls qdel, its right there in the name.  Its a macro and what you seem to have done here is broken it back up into the two lines it calls